code/game/objects/effects/spawners/mess_spawners.dm

```diff
@@ -0,0 +1,207 @@
+/obj/effect/spawner/themed_mess
+	name = "mess spawner"
+	icon = 'icons/effects/spawner_icons.dmi'
+	icon_state = "questionmark"
+
+	var/remaining_tiles
+
+/obj/effect/spawner/themed_mess/proc/drop_mess()
+	return
+
+/obj/effect/spawner/themed_mess/proc/drop_object()
+	return
+
+/obj/effect/spawner/themed_mess/proc/make_next_spawner(turf/T)
+	new type(T, remaining_tiles - 1)
+
+/obj/effect/spawner/themed_mess/New(loc, size = 10)
+	. = ..()
+	remaining_tiles = size
+	var/turf/T = get_turf(src)
+	if(!T)
+		stack_trace("Spawner placed in nullspace!")
+		return
+	randspawn(T)
+
+/obj/effect/spawner/themed_mess/proc/randspawn(turf/T)
+	drop_mess(T)
+
+	if(prob(25))
+		drop_object(T)
+
+	if(remaining_tiles > 0)
+		var/dirs = GLOB.alldirs.Copy()
+		shuffle_inplace(dirs)
+		for(var/dir in dirs)
+			var/turf/next = get_step(src, dir)
+			// Just check for plasteel turf to avoid walls and areas under windows
+			// findEventArea excludes maintenance anyway so we don't care about missing plating
+			if(istype(next, /turf/simulated/floor/plasteel) && !(locate(/obj/effect/decal/cleanable) in next))
+				make_next_spawner(next)
+				break
+	qdel(src)
+
+/obj/effect/spawner/themed_mess/party
+	name = "party mess spawner"
+
+/obj/effect/spawner/themed_mess/party/drop_mess(turf/T)
+	if(prob(50))
+		new /obj/effect/decal/cleanable/vomit(T)
+	else
+		new /obj/effect/decal/cleanable/confetti(T)
+
+/obj/effect/spawner/themed_mess/party/drop_object(turf/T)
+	if(prob(50))
+		new /obj/item/cigbutt(T) // does its own random placement transforms
+	else if(prob(20))
+		new /obj/item/cigbutt/roach(T)
+	else
+		var/trash_type = pick(
+			/obj/item/broken_bottle,
+			/obj/item/shard,
+		)
+		var/obj/trash = new trash_type(T)
+		trash.pixel_x = rand(-10,10)
+		trash.pixel_y = rand(-10,10)
+		trash.transform = turn(transform,rand(0,360))
+
+/obj/effect/spawner/themed_mess/bloody
+	name = "bloody mess spawner"
+	var/bloodcolor
+
+/obj/effect/spawner/themed_mess/bloody/New(loc, size = 10, blood = null)
+	bloodcolor = blood ? blood : pick(
+		"#004400",
+		"#0064c8",
+		"#1d2cbf",
+		"#2299fc",
+		"#a10808",
+		"#a200ff",
+		"#a3d4eb",
+		"#b9ae9c",
+		"#fb9800")
+	. = ..()
+
+/obj/effect/spawner/themed_mess/bloody/make_next_spawner(turf/T)
+	new type(T, remaining_tiles - 1, bloodcolor)
+
+/obj/effect/spawner/themed_mess/bloody/drop_mess(turf/T)
+	var/blood_decal_type = pick(
+		/obj/effect/decal/cleanable/blood,
+		/obj/effect/decal/cleanable/blood/splatter,
+		/obj/effect/decal/cleanable/blood/drip,
+		/obj/effect/decal/cleanable/blood/gibs,
+	)
+	var/obj/effect/decal/cleanable/blood/decal = new blood_decal_type(T)
+	decal.basecolor = bloodcolor
+	decal.update_icon()
+
+	if(prob(5))
```

Comment:
just a fwi, im fairly sure these aren't cleanable/removable. iirc the robotic ones are, but only by the drones

Reply:
me when wrong line
line 100, the remains, skull and crossbones thing
![image](https://github.com/ParadiseSS13/Paradise/assets/96908085/6e29ef82-6165-4b91-a6d3-39aad75242ee)

Reply:
You can't clean remains, yeah, it sucks

Reply:
sad. removed this for now.